scss/leaflet-sidebar.scss

```diff
@@ -10,76 +10,118 @@ $tab-hover-bg: #eee !default;
+    border-radius: $sidebar-border-radius;
 
     &.leaflet-touch {
-        box-shadow: none;
-        border-right: $sidebar-touch-border;
+      border: $sidebar-touch-border;
+    }
+  }
+}
+
+// move other controls aside from sidebar
+.leaflet-sidebar-left {
+  &.leaflet-touch {
+    box-shadow: none;
+    border-right: $sidebar-touch-border;
+  }
+
+  & ~ .leaflet-control-container .leaflet-left {
+    @media (min-width: $threshold-sm) {
+      transition: left $sidebar-transition;
     }
 
-    @media(min-width:$threshold-sm) {
-        border-radius: $sidebar-border-radius;
+    @media (min-width: $threshold-sm) and (max-width: $threshold-md - 1px) {
+      left: $width-sm + $sidebar-margins;
+    }
+
+    @media (min-width: $threshold-md) and (max-width: $threshold-lg - 1px) {
+      left: $width-md + $sidebar-margins;
+    }
 
-        &.leaflet-touch {
-            border: $sidebar-touch-border;
-        }
+    @media (min-width: $threshold-lg) {
+      left: $width-lg + $sidebar-margins;
     }
+  }
+
+  &.collapsed ~ .leaflet-control-container .leaflet-left {
+    @media (min-width: $threshold-sm) {
+      left: $tab-size + $sidebar-margins;
+    }
+  }
 }
 
-// move other controls aside from sidebar
-.leaflet-sidebar-left {
+.leaflet-sidebar-right {
+  &.leaflet-touch {
+    box-shadow: none;
+    border-left: $sidebar-touch-border;
+  }
+
+  & ~ .leaflet-control-container .leaflet-right {
+    @media (min-width: $threshold-sm) {
+      transition: right $sidebar-transition;
+    }
 
-    & ~ .leaflet-control-container .leaflet-left {
-        @media(min-width:$threshold-sm) {
-            transition: left $sidebar-transition;
-        }
+    @media (min-width: $threshold-sm) and (max-width: $threshold-md - 1px) {
+      right: $width-sm + $sidebar-margins;
+    }
 
-        @media(min-width:$threshold-sm) and (max-width:$threshold-md - 1px) {
-            left: $width-sm + $sidebar-margins;
-        }
+    @media (min-width: $threshold-md) and (max-width: $threshold-lg - 1px) {
+      right: $width-md + $sidebar-margins;
+    }
 
-        @media(min-width:$threshold-md) and (max-width:$threshold-lg - 1px) {
-            left: $width-md + $sidebar-margins;
-        }
+    @media (min-width: $threshold-lg) {
+      right: $width-lg + $sidebar-margins;
+    }
+  }
 
-        @media(min-width:$threshold-lg) {
-            left: $width-lg + $sidebar-margins;
-        }
+  &.collapsed ~ .leaflet-control-container .leaflet-right {
+    @media (min-width: $threshold-sm) {
+      right: $tab-size + $sidebar-margins;
     }
 
-    &.collapsed ~ .leaflet-control-container .leaflet-left {
-        @media(min-width:$threshold-sm) {
-            left: $tab-size + $sidebar-margins;
-        }
+    @media (max-width: $threshold-sm) {
+      right: $tab-size + $sidebar-margins;
     }
+  }
 }
 
-.leaflet-sidebar-right {
+.leaflet-sidebar-left {
+  &.leaflet-touch {
+    box-shadow: none;
+    border-left: $sidebar-touch-border;
```

Comment:
This should probably `border-right: 2px...`. Might fix the left border issue?

Reply:
True, now I cleaned the scss file from unnecessary lines.